src/main/java/org/galatea/starter/service/IexService.java

```diff
@@ -59,7 +68,42 @@
   public List<IexHistoricalPrices> getHistoricalPricesForSymbols(final String symbols,
       final String range, final String date) {
 
-    return iexHistoricalClient.getHistoricalPricesForSymbols(symbols, range, date);
+    String dateFormatted = date.substring(0,4) + "-" + date.substring(4, 6)
+        + "-" + date.substring(6, 8);
+
+    HistoricalPricesId historicalPricesId =
+        new HistoricalPricesId(symbols, dateFormatted, "10:50");
+
+    if (historicalPricesService.exists(historicalPricesId)) {
+
```

Comment:
Try and break this up into helper methods. It makes it easier to understand what each chunk of code does